### ISSUE TITLE
Accept bind-mounts with SELinux labelling via `-Z`/`-z` for Containerfiles

### ISSUE THREAD

**Issue by fbruetting** [hadolint/hadolint#974]
- [x] This is a bug report
- [ ] This is a feature request
- [x] I searched existing issues before opening this one

### Expected behavior

For Containerfiles, bind mounts with `-z` and `-Z` are possible (SELinux labelling.

### Actual behavior

Hadolint expects Dockerfile/Docker and throws an error:
```
unexpected "Z" expecting "from=", "readonly", "readonly=", "readwrite", "readwrite=", "ro", "ro=", "rw", "rw=", source=, or target=
```

### Steps to reproduce the behavior

Include a bind-mount `RUN` command like `RUN --mount=type=bind,Z,source=X,destination=Y <COMMAND>`.

**Output of `hadolint --version`:**

```bash
Haskell Dockerfile Linter 2.12.1
```

### Additionally

See Podman/Buildah documentation: https://docs.podman.io/en/latest/markdown/podman-build.1.html

**Comment by inean** [comment]
This issue is related to [language-docker#95](https://github.com/hadolint/language-docker/issues/95). We should  follow the discussion there.

**Comment by djhoese** [comment]
I'm running into this too. Since this isn't treated as a rule but rather a syntax error, is there a way to ignore it?

**Comment by eyupcanakman** [comment]
The bare `z`/`Z` flags would land in language-docker, since that's where the mount-arg parsing is. The existing `relabel=shared`/`relabel=private` support is in `Parser/Run.hs`, but it goes through the `relabel=` key, so the short form needs its own bare-flag branch.

Before I open a PR there, does mapping `z` to `RelabelShared` and `Z` to `RelabelPrivate` match what you had in mind? And should the pretty-printer keep `z`/`Z` as written, or normalize them back to the long `relabel=` form?

**Comment by m-ildefons** [comment]
> Before I open a PR there, does mapping z to RelabelShared and Z to RelabelPrivate match what you had in mind? And should the pretty-printer keep z/Z as written, or normalize them back to the long relabel= form?

Yes, that matches what the manual specifies: https://man.archlinux.org/man/Containerfile.5.en#FORMAT

Note, that much like `relabel=[shared|private]`, `z/Z` is only supported in *Containerfile* syntax (e.g. for Podman), not in [Dockerfile syntax](https://docs.docker.com/reference/dockerfile/#run---mounttypebind).
At some point this will need to be properly addressed, but for now I'm fine with just making the parser accept either.

**Comment by m-ildefons** [comment]
Fixed in v2.15.0
